### PR TITLE
Compiler Module Update

### PR DESCRIPTION
**Pull request by kiroco12**
Updated compiler module and tests passing in that module. Modules client, server and sbt-exercise haven't been updated, so they won't compile. The following tasks are to update the mentioned modules and after that, travis should pass the tests.

### CODE REVIEW

File: compiler/src/main/scala/org/scalaexercises/exercises/compiler/Compiler.scala [hunk 1]
```diff
@@ -124,18 +123,20 @@ case class Compiler() {
...
-        library: Library
-    ) =
+    def maybeMakeLibraryInfo(library: Library) =
       for {
         symbol ← internal.instanceToClassSymbol(library)
         symbolPath = internal.symbolToPath(symbol)
-        comment ← (internal.resolveComment(symbolPath) >>= Comments.parseAndRender[Mode.Library])
-          .leftMap(enhanceDocError(symbolPath) _)
-        sections ← checkEmptySectionList(symbol, library) >>= {
+        comment ← (internal
```
Comment (AdrianRaFo) [inline compiler/src/main/scala/org/scalaexercises/exercises/compiler/Compiler.scala]: we don't want unicode symbols

File: compiler/src/main/scala/org/scalaexercises/exercises/compiler/Compiler.scala [hunk 2]
```diff
@@ -337,16 +338,16 @@ case class Compiler() {
...
-        buildInfoT = buildInfoTerm
+        libraryInfo.comment.name,
+        libraryInfo.comment.description,
+        libraryInfo.color,
+        libraryInfo.logoPath,
+        libraryInfo.logoData,
+        sectionTerms,
+        libraryInfo.owner,
+        libraryInfo.repository,
+        compilationTimestamp,
```
Comment (AdrianRaFo) [inline compiler/src/main/scala/org/scalaexercises/exercises/compiler/Compiler.scala]: I would like to keep the params name here

File: compiler/src/main/scala/org/scalaexercises/exercises/compiler/Compiler.scala [hunk 3]
```diff
@@ -310,13 +311,13 @@ case class Compiler() {
...
-              imports = sectionInfo.imports,
-              path = sectionInfo.path,
-              contributionTerms = contributionTerms
+              libraryInfo.comment.name,
+              sectionInfo.comment.name,
+              sectionInfo.comment.description,
+              exerciseTerms,
+              sectionInfo.imports,
+              sectionInfo.path,
+              contributionTerms
```
Comment (AdrianRaFo) [inline compiler/src/main/scala/org/scalaexercises/exercises/compiler/Compiler.scala]: keep params name

File: compiler/src/main/scala/org/scalaexercises/exercises/compiler/Compiler.scala (hunk 1 above)
Comment (calvellido) [reply compiler/src/main/scala/org/scalaexercises/exercises/compiler/Compiler.scala]: Just out of curiosity @kiroco12, was this caused by some IntelliJ plugin or what?

Comment (kiroco12) [reply compiler/src/main/scala/org/scalaexercises/exercises/compiler/Compiler.scala]: No @calvellido, the code compiles with the unicode characters. But Adri prefer to use '=>' or '<-'

Comment (calvellido) [reply compiler/src/main/scala/org/scalaexercises/exercises/compiler/Compiler.scala]: But do you write them yourself with some key combination or what? 🤯

If you want the best of both worlds, you can us e afont with ligatures, like Fira Code:

https://github.com/tonsky/FiraCode

Comment (juanpedromoreno) [reply compiler/src/main/scala/org/scalaexercises/exercises/compiler/Compiler.scala]: That's something related to scalariform, I wouldn't care about that for now. Replace scalariform by scalafmt is probably a different task.

File: compiler/src/main/scala/org/scalaexercises/exercises/compiler/SourceTextExtraction.scala [hunk 4]
```diff
@@ -180,75 +180,75 @@ object SourceTextExtraction {
...
+      trees0 = List(Nil -> rootTree),
       acc0 = RawAcc[g.type](Nil, Nil),
-      visitDocComment = { (path, comment, acc) ⇒
+      visitDocComment = { (path, comment, acc) =>
         acc.copy(
           comments = (path, (acc.position, comment)) :: acc.comments,
           position = acc.position + 1
         )
-      //RawAcc(comments = (path → comment) :: Nil)
+      //RawAcc(comments = (path -> comment) :: Nil)
```
Comment (AdrianRaFo) [inline compiler/src/main/scala/org/scalaexercises/exercises/compiler/SourceTextExtraction.scala]: :fire:

File: compiler/src/main/scala/org/scalaexercises/exercises/compiler/SourceTextExtraction.scala [hunk 5]
```diff
@@ -180,75 +180,75 @@ object SourceTextExtraction {
...
       },
-      visitMethodExpr = { (path, expr, acc) ⇒
+      visitMethodExpr = { (path, expr, acc) =>
         acc.copy(
           methods = (path, (acc.position, expr)) :: acc.methods,
           position = acc.position + 1
         )
-      //RawAcc(methods = (path → expr) :: Nil)
+      //RawAcc(methods = (path -> expr) :: Nil)
```
Comment (AdrianRaFo) [inline compiler/src/main/scala/org/scalaexercises/exercises/compiler/SourceTextExtraction.scala]: fire

File: compiler/src/test/scala/org/scalaexercises/exercises/compiler/SourceTextExtractionSpec.scala [hunk 6]
```diff
@@ -85,24 +85,24 @@ class SourceTextExtractionSpec extends FunSpec with Matchers with Inside {
...
     }
 
     /*
     it("should capture imports at static scopes") {
-      res.imports.map { case (k, v) ⇒ k.mkString(".") → v.imports.map(renderImport).mkString(";") } should equal(Map(
-        "myPackage.Bar" → "Seqq.{ empty }",
-        "" → "scala.collection.immutable.{ Seq => Seqq }"
+      res.imports.map { case (k, v) => k.mkString(".") -> v.imports.map(renderImport).mkString(";") } should equal(Map(
+        "myPackage.Bar" -> "Seqq.{ empty }",
+        "" -> "scala.collection.immutable.{ Seq => Seqq }"
```
Comment (AdrianRaFo) [inline compiler/src/test/scala/org/scalaexercises/exercises/compiler/SourceTextExtractionSpec.scala]: :fire: ??

**Comment by kiroco12** [comment]
With Juan Pedro's approval and the resolved comments I'm going to merge this branch. Thanks to the reviewers.

**Pull request merged**